ui/src/criteria/attribute.tsx

```diff
@@ -81,73 +136,216 @@ function AttributeEdit(props: AttributeEditProps) {
+    const [minInputValue, setMinInputValue] = useState<number | string>(
+      props.data.min || minBound
+    );
+    const [maxInputValue, setMaxInputValue] = useState<number | string>(
+      props.data.max || maxBound
+    );
+
+    const updateValue = (newMin: number, newMax: number) => {
+      props.dispatchFn(
+        produce(props.data, (data) => {
+          data.min = newMin;
+          data.max = newMax;
+        })
+      );
+    };
+
+    const handleChange = (event: Event, newValue: number | number[]) => {
+      newValue = newValue as number[];
+      const [newMin, newMax] = newValue;
+      setMinInputValue(newMin);
+      setMaxInputValue(newMax);
+      updateValue(newMin, newMax);
+    };
+
+    const handleMinInputChange = (
+      event: React.ChangeEvent<HTMLInputElement>
+    ) => {
+      let newMin = event.target.value === "" ? "" : Number(event.target.value);
+      setMinInputValue(newMin); // Make sure empty input won't get changed.
+      newMin = typeof newMin === "number" ? newMin : 0;
+      const maxValue = props.data?.max || maxBound;
 
-  const selectionIndex = (hint: tanagra.EnumHintValue) =>
-    props.data.selected.findIndex(
-      (row) => row.id === hint.attributeValue?.int64Val
+      newMin = Math.max(minBound, newMin);
+      newMin = Math.min(maxValue, newMin);
+      updateValue(newMin, maxValue);
+    };
+    const handleMinInputBlur = () => {
+      let newMin = typeof minInputValue === "number" ? minInputValue : 0;
+      const maxValue = props.data?.max || maxBound;
+      newMin = Math.max(newMin, minBound);
+      newMin = Math.min(newMin, maxValue);
+      setMinInputValue(newMin);
+    };
+
+    const handleMaxInputChange = (
+      event: React.ChangeEvent<HTMLInputElement>
+    ) => {
+      let newMax = event.target.value === "" ? "" : Number(event.target.value);
+      setMaxInputValue(newMax); // Make sure empty input won't get changed.
+      newMax = typeof newMax === "number" ? newMax : 0;
+      const minValue = props.data?.min || minBound;
+
+      newMax = Math.max(minValue, newMax);
+      newMax = Math.min(newMax, maxBound);
+      updateValue(minValue, newMax);
+    };
+    const handleMaxInputBlur = () => {
+      let newMax = typeof maxInputValue === "number" ? maxInputValue : 0;
+      const minValue = props.data?.min || minBound;
+      newMax = Math.max(minValue, newMax);
+      newMax = Math.min(newMax, maxBound);
+      setMaxInputValue(newMax);
+    };
+
+    return (
+      <Box sx={{ width: "30%", minWidth: 300, margin: 2 }}>
+        <Grid container spacing={2}>
+          <Grid item>
+            <Input
+              value={minInputValue}
+              size="medium"
+              onChange={handleMinInputChange}
+              onBlur={handleMinInputBlur}
+              inputProps={{
+                step: ((maxBound - minBound) / 20) | 0,
+                min: minBound,
+                max: maxBound,
+                type: "number",
```

Comment:
According to the MUI docs you shouldn't use `type: "number"`. https://mui.com/material-ui/react-text-field/#type-quot-number-quot has an alternative.

Reply:
Hi @tjennison-google this does not work really well as our expectation. like the picture shows the left side did not have button on the input blank. 
<img width="646" alt="Screen Shot 2022-04-26 at 4 45 37 PM" src="https://user-images.githubusercontent.com/100877272/165389300-20100408-6b66-4ca9-a570-97641f45a85c.png">